crates/pallet-executor/src/lib.rs

```diff
@@ -24,18 +24,20 @@ use sp_executor::{
     SignedOpaqueBundle,
 };
 use sp_runtime::RuntimeAppPublic;
+use subspace_core_primitives::BlockNumber;
```

Comment:
Why is concrete type needed here if this is the type of primary chain anyway?

Reply:
Given that we have used the concrete type for `primary_hash` in `ExecutionReceipt` https://github.com/subspace/subspace/blob/b2072343cd92f7976fa53180598241dc696dcaa9/crates/sp-executor/src/lib.rs#L180, I also used the concrete type `BlockNumber` for `primary_number` in `ExecutionReceipt`, not sure it's better to use generic for `primary_number`.

Reply:
That data structure is *probably* fine, I was primarily talking about events and storage items, I'd use generic there as well, otherwise it is a weird mix. But I don't have a very strong opinion there.

Reply:
The events and storage items are from and for ExecutionReceipt, so it's fine by me. A cleaner way might be reworking the data structure ExecutionReceipt to use generic to potentially reduce the convention between concrete type and generic. I was thinking about this but haven't started, I can note a TODO if it looks reasonable.

Reply:
TODO added in https://github.com/subspace/subspace/pull/427/commits/4f63a0332d137b43c4b2729f57c7b5344578719d